vispy/visuals/pseudo_spheres.py

```diff
@@ -0,0 +1,190 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright (c) Vispy Development Team. All Rights Reserved.
+# Distributed under the (new) BSD License. See LICENSE.txt for more info.
+# -----------------------------------------------------------------------------
+"""Pseudo spheres (shaded discs) visual"""
+
+import numpy as np
+
+from ..color import ColorArray
+from ..gloo import VertexBuffer
+from .visual import Visual
+
+
+vert = """
+uniform vec3 u_light_position;
+
+attribute vec3 a_position;
+attribute vec4 a_color;
+attribute float a_radius;
+
+varying vec4 v_color;
+varying vec3 v_light_direction;
+varying float v_depth_middle;
+
+void main (void) {
+    v_color = a_color;
+    v_light_direction = normalize(u_light_position);
+
+    vec4 pos = vec4(a_position, 1.0);
+    vec4 fb_pos = $visual_to_framebuffer(pos);
+    gl_Position = $framebuffer_to_render(fb_pos);
+
+    // NOTE: gl_stuff uses framebuffer coords!
+
+    // calculate point size from visual to framebuffer coords to determine radius
+    vec4 x = $framebuffer_to_visual(fb_pos + vec4(100, 0, 0, 0));
+    x = (x/x.w - pos) / 100;
+    vec4 radius = $visual_to_framebuffer(pos + normalize(x) * a_radius);
+    radius = radius/radius.w - fb_pos/fb_pos.w;
+    // gl_PointSize is the diameter
+    gl_PointSize = length(radius) * 2;
```

Comment:
I think we should keep the size passed in as diameter, as that's how it is defined in `Markers`. Perhaps just rename to diameter or size.

If you want to use radius, I think need to remove the 0.5 in the depth calculation.

Reply:
I think you're right, and yet I can't for the life of me figure out why it works with both `*2` and `* 0.5` but not with just one. :sweat_smile: